src/filters/filter_lua_functions.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+/*
+ * This holds functions for the lua filter
+ */
+COMPILER_SUPRESS("-Wswitch-default")
+COMPILER_SUPRESS("-Weffc++")
+COMPILER_SUPRESS("-Wuseless-cast")
+COMPILER_SUPRESS("-Wswitch-enum")
+#include <sol/sol.hpp>
+COMPILER_RESTORE("-Wswitch-enum")
+COMPILER_RESTORE("-Wuseless-cast")
+COMPILER_RESTORE("-Weffc++")
+COMPILER_RESTORE("-Wswitch-default")
```

Comment:
how to suppress warnings of an external library (only header)?

Reply:
Well, you can disable warnings completely but I would't recommend that, as the warnings above are due to poor coding style within `sol` but it could also throw warnings if the library is used the wrong way.